pkg/webhook/mutation/pod/handler/otlp/handle.go

```diff
@@ -1,30 +1,57 @@
 package otlp
 
 import (
+	"fmt"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/consts"
+	"github.com/Dynatrace/dynatrace-operator/pkg/otlp/exporterconfig"
+	"github.com/Dynatrace/dynatrace-operator/pkg/webhook/mutation/pod/annotations"
 	dtwebhook "github.com/Dynatrace/dynatrace-operator/pkg/webhook/mutation/pod/mutator"
+	"github.com/Dynatrace/dynatrace-operator/pkg/webhook/mutation/pod/secrets"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type Handler struct {
+	kubeClient client.Client
+	apiReader  client.Reader
+
 	envVarMutator            dtwebhook.Mutator
 	resourceAttributeMutator dtwebhook.Mutator
 }
 
 func New(
+	kubeClient client.Client,
+	apiReader client.Reader,
 	envVarMutator dtwebhook.Mutator,
 	resourceAttributeMutator dtwebhook.Mutator,
 ) *Handler {
 	return &Handler{
+		kubeClient:               kubeClient,
+		apiReader:                apiReader,
 		envVarMutator:            envVarMutator,
 		resourceAttributeMutator: resourceAttributeMutator,
 	}
 }
 
 func (h *Handler) Handle(mutationRequest *dtwebhook.MutationRequest) error {
 	if !mutationRequest.DynaKube.OTLPExporterConfiguration().IsEnabled() {
+		log.Debug("OTLP injection disabled", "podName", mutationRequest.PodName(), "namespace", mutationRequest.Namespace.Name)
+
 		return nil
 	}
 
 	if h.envVarMutator.IsEnabled(mutationRequest.BaseRequest) {
+		if !h.isInputSecretPresent(
+			mutationRequest,
+			exporterconfig.GetSourceConfigSecretName(mutationRequest.DynaKube.Name),
+			consts.OTLPExporterSecretName,
+		) {
+			log.Debug("required input secret not present, skipping OTLP injection", "podName", mutationRequest.PodName(), "namespace", mutationRequest.Namespace.Name)
+
+			return nil
+		}
```

Comment:
something is wrong here
during my test I managed to have it error
```
Error: secret "dynatrace-otlp-exporter-config" not found
```

How to reproduce:
- Have a `token` secret, that has no `dataIngestToken`
- So the Operator will correctly error, as it can't create the `<dk>-otlp-exporter-config`
- But the webhook still injects